File: 58-length-of-last-word/length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
        int size = s.length();
        int res = 0; 

        for(int i = 0; i < size; i++) {
            char currentLetter = s.charAt(size - 1 - i);
            if (currentLetter == ' ' && res == 0) {
                continue;
            }
            else if (currentLetter == ' ') {
                return res;
            } 
            res += 1;
        }
        return res;
    }
}

File: 58-length-of-last-word/length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        reverse_s = s[::-1]
        i, length = 0, 0

        while reverse_s[i] == ' ':
            i += 1

        while i < len(s) and reverse_s[i] != ' ':
            length += 1
            i += 1
        
        return length


        